test-app/tests/integration/components/combobox-test.gts

```diff
@@ -0,0 +1,854 @@
+
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'ArrowDown');
+
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'false');
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'true');
+  });
+
+  test('it sets the "active" item to the previous item in the list when using the UP arrow', async function (assert) {
+    // NOTE: Setting the selected option to "red" here so that
+    // the last item in the list will be active so that we can
+    // move up!
+    await render(<template>
+      <ComboboxControl
+        @selected="red"
+        @options={{testColors}}
+        data-combobox
+        as |combobox|
+      >
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    await click('[data-combobox]');
+
+    assert.dom('[role="option"]').exists({ count: 2 });
+
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'ArrowUp');
+
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'true');
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'false');
+  });
+
+  test('it closes an open popover when the ESCAPE key is pressed', async function (assert) {
+    await render(<template>
+      <ComboboxControl @options={{testColors}} data-combobox as |combobox|>
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    await click('[data-combobox]');
+
+    assert.dom('[role="listbox"]').exists();
+
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'Escape');
+
+    assert.dom('[role="listbox"]').doesNotExist();
+  });
+
+  test('it closes an open popover when the component is blurred', async function (assert) {
+    await render(<template>
+      {{! template-lint-disable require-input-label }}
+      <input placeholder="test" data-input />
+
+      <ComboboxControl @options={{testColors}} data-combobox as |combobox|>
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    // Open the popover
+    await click('[data-combobox]');
+
+    assert.dom('[role="listbox"]').exists();
+
+    // Now blur the elment
+    await click('[data-input]');
+
+    assert.dom('[role="listbox"]').doesNotExist();
+  });
+
+  test('it reopens the popover when any key is pressed if the popover is closed', async function (assert) {
+    await render(<template>
+      <ComboboxControl @options={{testColors}} data-combobox as |combobox|>
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    // Open the popover by clicking it
+    await click('[data-combobox]');
+
+    // Now close it
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'Escape');
+    // Now reopen it
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'ArrowDown');
```

Comment:
Super tiny nit / question: you can also reopen it with `Escape`, is this intentional? I actually don't mind this experience tho (less having to move around the keyboard)

Reply:
Yeah, it is intentional, but not sure if it's "correct".  I thought any keyboard input opening the popover was suitable UX, but that's maybe just a personal preference.  I'm glad you don't mind it either!